dm/syncer/error.go

```diff
@@ -61,6 +68,95 @@ func isDropColumnWithIndexError(err error) bool {
 			strings.Contains(mysqlErr.Message, "with tidb_enable_change_multi_schema is disable"))
 }
 
+// getDDLStatusFromTiDB retrieves the synchronizing status of DDL from TiDB
+// hence here db should be TiDB database
+// createTime should be based on the timezone of downstream, and its unit is second.
+func getDDLStatusFromTiDB(ctx context.Context, db *sql.DB, ddl string, createTime int64) (string, error) {
+	rowNum := 10
+	rowOffset := 0
+	queryMap := make(map[int]string)
+
+	for {
+		// every attempt try 10 history jobs
+		showJobs := fmt.Sprintf("ADMIN SHOW DDL JOBS %d", rowNum)
+		jobsRows, err := db.QueryContext(ctx, showJobs)
+		if err != nil {
+			return "", err
+		}
+
+		var jobsResults [][]string
+		jobsResults, err = export.GetSpecifiedColumnValuesAndClose(jobsRows, "JOB_ID", "CREATE_TIME", "STATE")
+		if err != nil {
+			return "", err
+		}
+
+		for i := rowNum - 10; i < rowNum; i++ {
+			ddlCreateTimeStr := jobsResults[i][1]
```

Comment:
Will lens of jobsResults small then 10?

Reply:
I don't think it's possible. I checked by initiating a new TiDB, do nothing then type ```ADMIN SHOW DDL JOBS```, and it returned 10 rows, it seems that every initialized TiDB would have 33 rows of history jobs by default. But to be safe, I added the situation where the lens of jobsResults is small than 10